Tools/clinic/clinic.py

```diff
@@ -872,47 +872,49 @@ def deprecate_positional_use(
             params: dict[int, Parameter],
     ) -> str:
         assert len(params) > 0
-        # FIXME: Handle multiple deprecation levels
-        # FIXME: For now, assume there's only one level
         code_blocks: list[str] = []
         names = [repr(p.name) for p in params.values()]
         first_pos, first_param = next(iter(params.items()))
+        last_pos, last_param = next(reversed(params.items()))
+        # FIXME: Handle multiple deprecation levels
+        # FIXME: For now, assume there's only one level
+        assert first_param.deprecated_positional == last_param.deprecated_positional
 
         # Pretty-print list of names.
         match len(params):
             case 1:
-                pstr = first_param.name
+                pstr = repr(first_param.name)
             case 2:
                 pstr = " and ".join(names)
             case _:
                 pstr = ", ".join(names[:-1]) + " and " + names[-1]
 
         # Format the preprocessor warning and error messages.
+        assert isinstance(self.cpp.filename, str)
         source = os.path.basename(self.cpp.filename)
         thenceforth = first_param.deprecated_positional  # FIXME
         assert thenceforth is not None
         major, minor = thenceforth
-        assert isinstance(self.cpp.filename, str)
         cpp_message = (
             f"In {source}, update parameter(s) {pstr} in the clinic "
             f"input of {func.full_name!r} to be keyword-only."
         )
         # Format the deprecation message.
-        singular = (
-            f"Passing {pstr} as a positional parameter is deprecated. "
-            f"It will become a keyword-only parameter in Python {major}.{minor}."
-        )
-        plural = (
-            f"Passing {pstr} as positional parameters is deprecated. "
-            f"They will become keyword-only parameters in Python {major}.{minor}."
-        )
+        depr_message = (f"Passing more than 1 positional arguments to "
```

Comment:
Of course `1` should not be hardcoded. It was only in the example function.

Reply:
Sorry, I'm trying to paint the house _and_ address review remarks at the same time. Turns out that is not my best idea 😄 I'll amend this.